modules/deploy/pages/deployment-option/cloud/gcp-private-service-connect.adoc

```diff
@@ -0,0 +1,425 @@
+= Configure GCP Private Service Connect for BYOC
+:description: Set up GCP Private Service Connect to securely access Redpanda Cloud. 
+:page-cloud: true
+
+include::shared:partial$feature-flag.adoc[]
+
+// This guide describes the steps for setting up Private Service Connect (PSC) on Redpanda Cloud.
+
+// Redpanda GCP PSC can be enabled during cluster creation or for an existing cluster.
+
+// After completing this guide, a PSC service attachment will have been deployed for your Redpanda Cloud cluster. You may then establish connections to it from any GCP project included in the allow list.
+
+After <<get-a-cloud-api-access-token,getting an access token>>, you can <<create-new-byovpc-cluster-with-psc-enabled,enable PSC when creating a new BYOC cluster>>, or you can <<enable-psc-on-an-existing-byovpc-cluster,enable PSC for existing BYOC clusters>>.
+
+== Requirements
+
+* You must use the Redpanda Cloud API to enable the Redpanda endpoint service for your clusters. Follow the steps below to <<get-a-cloud-api-access-token,get an access token>>. 
+
+== Get a Cloud API access token 
+
+. Save the base URL of the Redpanda Cloud API in an environment variable:
++
+[,bash]
+----
+PUBLIC_API_ENDPOINT="https://api.cloud.redpanda.com"
+----
+
+. In the https://cloudv2.redpanda.com/[Redpanda Cloud UI], go to **Namespaces** and select the namespace in which you want to create a cluster.
++
+Copy and store the namespace ID (UUID) from the URL in the browser.
++
+[,bash]
+----
+NAMESPACE_ID=<uuid>
+----
+
+. Go to the organization (org) level **Home** in the UI, and navigate to **Clients**. If you don't have an existing client, you can create a new one by clicking **Add client**.
++
+Copy and store the client ID and secret.
++
+[,bash]
+----
+CLOUD_CLIENT_ID=<client-ID>
+CLOUD_CLIENT_SECRET=<client-secret>
+----
+
+. Get an API token using the client ID and secret. You can click the **Request an API token** link to see code examples to generate the token.
++
+[,bash]
+----
+AUTH_TOKEN=`curl -s --request POST \
+    --url 'https://auth.prd.cloud.redpanda.com/oauth/token' \
+    --header 'content-type: application/x-www-form-urlencoded' \
+    --data grant_type=client_credentials \
+    --data client_id=$CLOUD_CLIENT_ID \
+    --data client_secret=$CLOUD_CLIENT_SECRET \
+    --data audience=cloudv2-production.redpanda.cloud | jq .access_token | sed 's/"//g'`
+----
+
+You must send the API token in the `Authorization` header when making requests to the Cloud API.
+
+== BYOVPC
```

Comment:
Is the "BYOVPC" terminology what we officially want to use in our documentation, or is this colloquial / internal only? I noticed that we don't have it anywhere in our docs just yet. Is "customer-managed VPC" (doc [here](https://docs.redpanda.com/current/deploy/deployment-option/cloud/vpc-byo-gcp/)) basically what we are referring to with BYOVPC?

Reply:
Oh, haha. I actually just sent you a message on Slack asking you the same question. Let me ask in #product.

Reply:
@kbatuigas: Per @bpraseed, we should use "customer-managed VPC" (not BYOVPC)